python/ray/tune/trainable.py

```diff
@@ -555,6 +555,16 @@ def iteration(self):
         """
         return self._iteration
 
+    @property
+    def training_iteration(self):
```

Comment:
why the duplicate functionality?

Reply:
because `training_iteration` shows up in the metrics, but there's no way to interact with `training_iteration`. We can remove this from this PR for now if you want

Reply:
up to you

does it make sense to deprecate `self.iteration` then?

Reply:
yeah, we can do that (though for now I wouldn't touch it in this PR, I can make a new issue)